### ISSUE TITLE
Incorrect use of Map type

### ISSUE THREAD

**Issue by rintaun** [FusionAuth/fusionauth-typescript-client#11]
## Problem

There are a number of types defined using `Map<K, V>`, but they are not actually instances of [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map), which leads to type warnings such as:

```
Element implicitly has an 'any' type because type 'Map<string, Error[]>' has no index signature. Did you mean to call 'get' ?
```

## Solution

As far as I can tell, these are all plain objects, so `Map<K, V>` should be replaced with `Record<K, V>`

**Comment by han4wluc** [comment]
I am attempting to register a user with data. But the data is always empty. I suspect this may be because the data is a Map type. Same thing if I try to use the `updateUser`, `patchUser` API.

```
import { FusionAuthClient } from '@fusionauth/typescript-client'

const client = new FusionAuthClient(
  'xxxx',
  'http://localhost:9011'
)

const main = async () => {
  try {
    const res = await client.register(undefined, {
      user: {
        username: 'bbbwer',
        password: 'rqergeq34aL',
        data: new Map([
          ['aaa', 'one'],
          ['bb', 'eee'],
          ['displayName', 'zERwje']
        ])
      },
      registration: {
        applicationId: '3149641a-81d8-4202-9715-7d5367ca94f1',
      }
    })

    console.warn('res', res)
  } catch (error) {
    console.warn('error', JSON.stringify(error, null, 2))
  }
}

main()
```


If I run the code. response is:
```
res ClientResponse {
  statusCode: 200,
  response: {
    registration: {
      applicationId: '3149641a-81d8-4202-9715-7d5367ca94f1',
      id: '0e29262e-fb33-4b13-8498-eccd22141f18',
      insertInstant: 1584104747639,
      usernameStatus: 'ACTIVE',
      verified: true
    },
    user: {
      active: true,
      id: '7abf7a10-13eb-4f26-8cb2-52338fdc7421',
      insertInstant: 1584104747566,
      passwordChangeRequired: false,
      passwordLastUpdateInstant: 1584104747601,
      tenantId: '3e75863b-17bd-712f-22c6-d884ba501412',
      twoFactorDelivery: 'None',
      twoFactorEnabled: false,
      username: 'bbbwer',
      usernameStatus: 'ACTIVE',
      verified: true
    }
  }
}
```

There is no data field.

<img width="622" alt="Screen Shot 2020-03-13 at 2 07 35 PM" src="https://user-images.githubusercontent.com/8134740/76623573-084a1900-6534-11ea-8663-ee260f1f0ac0.png">

**Comment by rintaun** [comment]
@han4wluc try using a normal object instead of a `Map`, e.g.

```js
{
  aaa: 'one',
  bb: 'eee',
  displayName: 'zERwje',
}
```

**Comment by han4wluc** [comment]
> @han4wluc try using a normal object instead of a `Map`, e.g.
> 
> ```js
> {
>   aaa: 'one',
>   bb: 'eee',
>   displayName: 'zERwje',
> }
> ```

Yes, I have tried, but this way I get a type error.

<img width="718" alt="Screen Shot 2020-03-13 at 2 26 13 PM" src="https://user-images.githubusercontent.com/8134740/76625010-a808a680-6536-11ea-9ce2-e988cf8c8af1.png">

**Comment by rintaun** [comment]
have you tried suppressing the error with `@ts-ignore`? if not, I would check to see if that works; that type error is incorrect because the `Map` type is being used incorrectly.

**Comment by han4wluc** [comment]
> have you tried suppressing the error with `@ts-ignore`? if not, I would check to see if that works; that type error is incorrect because the `Map` type is being used incorrectly.

Yes, it works with `@ts-ignore`. Thank you. I will use this workaround for now.

**Comment by robotdan** [comment]
Thanks @rintaun for the assist! 

@tyduptyler13 Is this something we can incorporate into our client library to make this work a bit better?

**Comment by tyduptyler13** [comment]
This is something I can change in our client builder type mapping file. I will look into fixing it.

**Comment by tyduptyler13** [comment]
Published the change in `1.15.8`. Let me know if you have any problems.

**Comment by rintaun** [comment]
@tyduptyler13 is there a particular reason for the choice to use a custom `ObjectMap<K,T>` type over the built-in `Record<K,T>`? It looks like its doing the same thing
